### PR TITLE
Rewards ads android

### PR DESCRIPTION
**Pull request by petemill**
Rewards feature supported for ads on android (settings page).

- BoxMobile changes to support an alert https://github.com/brave/browser-android-tabs/issues/1519
- BoxMobile changes to support 'extra (non-text) description content' https://github.com/brave/browser-android-tabs/issues/1580
- GrantComplete has props for custom text
- UGP / Ads grant changes required for https://github.com/brave/browser-android-tabs/issues/1582

Required by android branch at https://github.com/brave/brave-core/pull/2508

## Test plan
- Build is ok

##### Link / storybook path to visual changes
- https://brave-ui-1ijscynum.now.sh

## Integration
- [ ] Does this contain changes to src/components or src/
  - [ ] Will you publish to npm immediately after this PR, or wait until sometime in the future?
  - [ ] Incompatible API change to something existing _(major version increase)_
  - [ ] Adding new backwards-compatible functionality? _(minor version increase)_
  - [ ] Fixing a bug backwards-compatibly? _(patch version increase)_
  
- [x] Does this contain changes to src/features for brave-core?
  - [x] Are there non backwards-compatible changes required for brave-core? **Do not merge until brave-core PR is approvable.** Link to brave-core PR: https://github.com/brave/brave-core/pull/2509
  - [x] Will you create brave-core PR to update to this commit after it is merged?
  - [ ] Wants uplift to brave-core feature branch?
     - When uplift-approved, merge to brave-core-0.VV.x feature branch
     - Create additional brave-core PRs for each feature branch to update commit

### CODE REVIEW

File: src/features/rewards/mobile/boxMobile/index.tsx [hunk 1]
```diff
@@ -40,6 +12,7 @@ import {
 import { List } from '../../'
 import { getLocale } from '../../../../helpers'
 import Toggle from '../../../../components/formControls/toggle/index'
+import * as Styled from './style'
```
Comment (rossmoody) [inline src/features/rewards/mobile/boxMobile/index.tsx]: 🎉

**Comment by petemill** [comment]
Updated (deployment link included) to add GrantComplete customization

**Comment by emerick** [comment]
@petemill Just FYI, I pulled these changes into https://github.com/brave/brave-core/pull/3886.

**Comment by NejcZdovc** [comment]
closing as this was resolved

**Pull request closed without merging**